From 87ff0393af8762f396da97502f990dfbc57e1d32 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:42 -0400
Subject: [tools] perf/timechart: Always try to print at least 15 tasks

Message-id: <1408953210-26343-35-git-send-email-jolsa@redhat.com>
Patchwork-id: 88795
O-Subject: [PATCH RHEL7.1 BZ1133083 034/322] perf timechart: Always try to print at least 15 tasks
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 0a8eb275cbdb8462854d5f7e1168d86cee4cc9ea
Author: Stanislav Fomichev <stfomichev@yandex-team.ru>
Date: Fri Nov 1 20:25:45 2013 +0400

description
===========
Always try to print at least 15 tasks no matter how long they run.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-timechart.c b/tools/perf/builtin-timechart.c
index 41c9bde..bb21e57 100644
--- a/tools/perf/builtin-timechart.c
+++ b/tools/perf/builtin-timechart.c
@@ -945,15 +945,17 @@ static void write_svg_file(const char *filename)
 {
  u64 i;
  int count;
+ int thresh = TIME_THRESH;
 
  numcpus++;
 
 
- count = determine_display_tasks(TIME_THRESH);
-
- /* We'd like to show at least 15 tasks; be less picky if we have fewer */
- if (count < 15)
-  count = determine_display_tasks(TIME_THRESH / 10);
+ /* We'd like to show at least proc_num tasks;
+  * be less picky if we have fewer */
+ do {
+  count = determine_display_tasks(thresh);
+  thresh /= 10;
+ } while (!process_filter && thresh && count < 15);
 
  open_svg(filename, numcpus, count, first_time, last_time);
 
-- 
1.7.1